app/components/AddTasks.tsx

```diff
@@ -18,7 +18,7 @@ const AddTasks = () => {
 
   const handleAddTask = async () => {
     const newTask: Task = {
-      id: '', // generated by the task service
+      id: crypto.randomUUID(),
```

Comment:
Decided not to create these in services... but I'm not sure that's a great idea. 

Maybe these functions (from the store) should only take the data we're concerned with here on the frontend - the task/note text/content - abstracting things like `id`, dates, etc from needing to be passed from the frontend.